### PR TITLE
DB/Quest: Assassin's Contract" & "Baron's Demise

### PR DESCRIPTION
**Pull request by poszer**
- Fix for the Quest

Co-authored-by: Knindza <knindzagxg@users.noreply.github.com>

Fix for the Quest : Assasin's Contract & Baron's Demise

##### CHANGES PROPOSED:

-  Changed SAI, added coordinates for it
-  Added 4 Guards
- Assasin's will auto despawn after 60 sec when they re killed
- Respawn is diesbled until next event. 

###### ISSUES ADDRESSED:

Closes https://github.com/azerothcore/azerothcore-wotlk/issues/1461

##### TESTS PERFORMED:

Sql is imported without issue. Tested in game with @Knindzagxg and everything is fine. P.S This quest didn't work before.

##### HOW TO TEST THE CHANGES:

1. .tele Southshore
2. Check if there is Assasin or Criiera
3. reload creature_template for every id
4. reload  smart_sccripts
5. restart and tele there again 
 [Video here :](https://www.youtube.com/watch?v=mz6AmNLBiak&feature=youtu.be)

##### KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

##### Target branch(es):

Master

Big thanks for @Knindzagxg for fixing this event!!!

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1553383116284680300.sql [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+(2435, 0, 9, 0, 1, 0, 100, 1, 150000, 150000, 0, 0, 12, 2434, 6, 60000, 0, 0, 0, 8, 0, 0, 0, -805.856506, -479.743591, 15.871011, 5.608394, ''),
+(2435, 0, 10, 0, 1, 0, 100, 1, 152000, 152000, 0, 0, 12, 2434, 6, 60000, 0, 0, 0, 8, 0, 0, 0, -887.816284, -545.123413, 7.047425, 0.609322, ''),
+(2435, 0, 11, 0, 1, 0, 100, 1, 154000, 154000, 0, 0, 12, 2434, 6, 60000, 0, 0, 0, 8, 0, 0, 0, -893.224487, -590.699219, 7.445801, 1.06091, ''),
+(2435, 0, 12, 0, 1, 0, 100, 1, 156000, 156000, 0, 0, 1, 4, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, ''),
+(2435, 0, 13, 0, 11, 0, 100, 1, 0, 0, 0, 0, 41, 180000, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, '');
+
+INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `spawndist`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `ScriptName`, `VerifiedBuild`) VALUES (2054680, 2435, 0, 0, 0, 1, 1, 0, 1, -743.976, -521.674, 21.6875, 0.356982, 300, 0, 0, 2033, 0, 0, 0, 0, 0, '', 0);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1553383116284680300.sql]: missing `DELETE` before `INSERT`

File: data/sql/updates/pending_db_world/rev_1553383116284680300.sql [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+-- Guard 1
+INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `spawndist`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `ScriptName`, `VerifiedBuild`) VALUES (2054762, 2386, 0, 0, 0, 1, 1, 0, 1, -830.2, -530.016, 13.5798, 5.58997, 300, 0, 0, 2310, 0, 2, 0, 0, 0, '', 0);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1553383116284680300.sql]: missing `DELETE` before `INSERT`

File: data/sql/updates/pending_db_world/rev_1553383116284680300.sql [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+INSERT INTO `creature_addon` (`guid`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `auras`) VALUES (2054762, 20547620, 0, 0, 0, 0, NULL);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1553383116284680300.sql]: missing `DELETE` before `INSERT`

File: data/sql/updates/pending_db_world/rev_1553383116284680300.sql [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547580, 3, -852.599, -541.857, 10.5971, 0, 0, 0, 0, 100, 0);
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547580, 2, -856.708, -542.947, 10.352, 0, 0, 0, 0, 100, 0);
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547580, 1, -860.698, -518.619, 10.9982, 0, 0, 0, 0, 100, 0);
+
+-- Guard 4
+INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `spawndist`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `ScriptName`, `VerifiedBuild`) VALUES (2054765, 2386, 0, 0, 0, 1, 1, 0, 1, -894.552, -593.034, 8.03216, 0.822596, 300, 0, 0, 2310, 0, 0, 0, 0, 0, '', 0);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1553383116284680300.sql]: missing `DELETE` before `INSERT` in all above lines

File: data/sql/updates/pending_db_world/rev_1553383116284680300.sql (hunk 3 above)
Comment (Nefertumm) [reply data/sql/updates/pending_db_world/rev_1553383116284680300.sql]: Still missing `DELETE`.

File: data/sql/updates/pending_db_world/rev_1553383116284680300.sql [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547620, 4, -811.54, -537.426, 15.5707, 0, 0, 0, 0, 100, 0);
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547620, 3, -812.723, -542.349, 15.7681, 0, 0, 0, 0, 100, 0);
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547620, 2, -818.293, -536.134, 15.4268, 0, 0, 0, 0, 100, 0);
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547620, 1, -824.617, -532.774, 14.6419, 0, 0, 0, 0, 100, 0);
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547620, 6, -827.359, -531.308, 14.2184, 0, 0, 0, 0, 100, 0);
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1553383116284680300.sql]: Make one query for these lines. 
```sql
INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547620, 5, -820.391, -533.587, 15.0998, 0, 0, 0, 0, 100, 0),
(20547620, 4, -811.54, -537.426, 15.5707, 0, 0, 0, 0, 100, 0),
(20547620, 3, -812.723, -542.349, 15.7681, 0, 0, 0, 0, 100, 0),
(20547620, 2, -818.293, -536.134, 15.4268, 0, 0, 0, 0, 100, 0),
(20547620, 1, -824.617, -532.774, 14.6419, 0, 0, 0, 0, 100, 0),
(20547620, 6, -827.359, -531.308, 14.2184, 0, 0, 0, 0, 100, 0);
```
Also, missing `DELETE` before `INSERT` too.

File: data/sql/updates/pending_db_world/rev_1553383116284680300.sql [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547900, 4, -801.21, -563.578, 15.4117, 0, 0, 0, 0, 100, 0);
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547900, 5, -803.931, -564.6, 15.2881, 0, 0, 0, 0, 100, 0);
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547900, 6, -801.668, -574.03, 15.2231, 0, 0, 0, 0, 100, 0);
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547900, 7, -803.813, -581.047, 15.1528, 0, 0, 0, 0, 100, 0);
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547900, 8, -816.032, -584.642, 15.1528, 0, 0, 0, 0, 100, 0);
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1553383116284680300.sql]: Same as above.

File: data/sql/updates/pending_db_world/rev_1553383116284680300.sql [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547580, 5, -839.172, -527.138, 11.2478, 0, 0, 0, 0, 100, 0);
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547580, 4, -854.201, -532.857, 9.89841, 0, 0, 0, 0, 100, 0);
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547580, 3, -852.599, -541.857, 10.5971, 0, 0, 0, 0, 100, 0);
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547580, 2, -856.708, -542.947, 10.352, 0, 0, 0, 0, 100, 0);
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES (20547580, 1, -860.698, -518.619, 10.9982, 0, 0, 0, 0, 100, 0);
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1553383116284680300.sql]: Same.

**Comment by Nefertumm** [comment]
Still, concatenate the queries.

**Comment by FrancescoBorzi** [comment]
http://www.azerothcore.org/wiki/Dealing-with-SQL-files#write-compact-code

**Comment by poszer** [comment]
Tested in game with @Knindzagxg and everything is fine. P.S This quest didn't work before. Now work perfectly! Can be merged :+1:

**Comment by jorgeEF** [comment]
Assassin's won't stop spawning on my server and became an army that kills all the people in southshore. How to disable this?

**Pull request merged**